app/views/api_keys/new.html.erb

```diff
@@ -5,11 +5,27 @@
     <%= label_tag :name, t("api_keys.index.name"), class: "form__label" %>
     <%= f.text_field :name, class: "form__input", autocomplete: :off %>
 
-    <%= label_tag :scope, t("api_keys.index.scope"), class: "form__label" %>
-    <% ApiKey::API_SCOPES.each do |api_scope| %>
-      <div class = "form__checkbox__item">
-        <%= f.check_box "#{api_scope}", { class: "form__checkbox__input", id: "#{api_scope}" } , "true", "false" %>
-        <%= label_tag api_scope, t("api_keys.index.#{api_scope}"), class: "form__checkbox__label" %>
+    <div class="form__group">
+      <%= label_tag :scope, t("api_keys.index.scope"), class: "form__label" %>
+      <% ApiKey::API_SCOPES.each do |api_scope| %>
+        <div class = "form__checkbox__item">
+          <%= f.check_box "#{api_scope}", { class: "form__checkbox__input", id: "#{api_scope}" } , "true", "false" %>
+          <%= label_tag api_scope, t("api_keys.index.#{api_scope}"), class: "form__checkbox__label" %>
+        </div>
+      <% end %>
+    </div>
+
+    <% if current_user.mfa_enabled? %>
+      <div class="form__group">
+        <%= label_tag :mfa, t("api_keys.index.multifactor_auth"), class: "form__label" %>
+        <% if current_user.mfa_ui_and_api? %>
+          <p><%= t("api_keys.index.mfa_api_enabled") %></p>
+          <%= f.hidden_field :mfa, value: true %>
```

Comment:
when a user has mfa_ui_and_api, all keys they create will have mfa enabled. is this desired behavior? this seems inconsistent with existing keys, where mfa is null for user with mfa_ui_and_api. when user enables mfa with ui_only, it may be unexpected that his existing keys are not working without otp.
also, if we really want this to be true, we should set this in the controller and not view (attackers can mess with this logic).

Reply:
My thought process is that a user that has mfa ui and api enabled would downgrade if one of their keys can't work with otp. It would be a pain to downgrade and enable mfa on most of their existing keys if they have many.

I agree that the inconsistencies between new/existing keys isn't great. On the new page, we are communicating this inconsistency by saying mfa will be turned on for that key. One possibility to try to make it more consistent (other than backfilling) is to set mfa to true when updating existing keys if the user has mfa_ui_and_api. But that could make it even more confusing if the user downgrades. 

I removed the hidden value that sets this in the view and will add it to the controller if we decide to go this way.

Reply:
>  to make it more consistent (other than backfilling) is to set mfa to true when updating existing keys if the user has mfa_ui_and_api.

agreed, we can make this consistent.

> My thought process is that a user that has mfa ui and api enabled would downgrade if one of their keys can't work with otp.

got it. thank you for explaining. Perhaps we need more opinion on this, however, I feel if the user is changing their mfa level, they would expect that existing keys would use the default behavior for that level (default mfa false for ui_only). we would preserve api_key.mfa only where it was explicitly enabled. 

If we agree on the above proposition, setting api_key.mfa, when user has mfa UI and API enabled, would be redundant.  mfa will be enforced anyway for all keys.

Reply:
> Perhaps we need more opinion on this, however, I feel if the user is changing their mfa level, they would expect that existing keys would use the default behavior for that level (default mfa false for ui_only). we would preserve api_key.mfa only where it was explicitly enabled.

That makes a lot of sense, I agree with your view now. I’m happy to leave it the way it is right now and not set `api_key.mfa` unless the user explicitly sets it in the ui only level. I suppose it also isn’t difficult to change the behaviour if we change our minds about this in the future?

Reply:
yes, I would prefer that we don't set mfa when the user already has MFA UI and api.

Reply:
👍 I removed the checkbox associated with enabling MFA on the UI and API level.

<img width="960" alt="Screen Shot 2021-11-29 at 9 31 07 AM" src="https://user-images.githubusercontent.com/42748004/143886153-7e54fcb1-e51e-4247-b297-a57fb1b3da42.png">